ironic_prometheus_exporter/messaging.py

```diff
@@ -18,13 +19,32 @@ class PrometheusFileDriver(notifier.Driver):
     """Publish notifications into a File to be used by Prometheus"""
 
     def __init__(self, conf, topics, transport):
-        self.file_path = conf.oslo_messaging_notifications.file_path
-        if not self.file_path.endswith('.json'):
-            raise Exception('The file should end with .json')
-        if not os.path.exists(os.path.dirname(self.file_path)):
-            os.makedirs(os.path.dirname(self.file_path))
+        self.files_dir = conf.oslo_messaging_notifications.files_dir
```

Comment:
nit: caching the option value like that will prevent it from becoming updateable on SIGHUP. Not sure if you want it, of course..

Reply:
by 'prevent it from becoming updateable' you mean if someone change the configuration and restart the service we wont have the new value?

Reply:
I mean, without restarting, with just sending SIGHUP

Reply:
Not sure the correct way to do this, ideas are welcome =)

Reply:
In the environment where this is used we don't need to worry about SIGHUP for restarts.